### DISCUSSION TITLE
Dell 15R-5537 (4th gen i5) Updating past 115 fails to load

### DISCUSSION THREAD

**Discussion by madhits** [sebanc/brunch#1938] [Q&A]
Hello,

I have Been running chrome OS on the dell inspiron 15r for a while, I recently went to update to 116 and 117 both fail to load. A new USB built with 117 fails to even boot up and freezes. I can provide screen shots of the issue but it usually ends up freezing due to my elan touch screen, that's the point the boot up wont pass.  I have an install on the system that is between updating from 115 and 116 and I've tried all of the framework options to no avail. Hope someone has some ideas, the 15r is very compatible along with a number of dell variants so this is a real surprise to me. I'm using Rammus recovery specifically: https://cros.tech/device/shyvana/  @WesBosch @tejasraman

**Comment by tejasraman** [comment]
> Hello,
> 
> I have Been running chrome OS on the dell inspiron 15r for a while, I recently went to update to 116 and 117 both fail to load. A new USB built with 117 fails to even boot up and freezes. I can provide screen shots of the issue but it usually ends up freezing due to my elan touch screen, that's the point the boot up wont pass.  I have an install on the system that is between updating from 115 and 116 and I've tried all of the framework options to no avail. Hope someone has some ideas, the 15r is very compatible along with a number of dell variants so this is a real surprise to me.  @WesBosch @tejasraman 

I'll try to help as much as possible, but I need to know whether it's a 15R 5521 or 5537 and what recovery image you're using, as well as the kernel version you're using.

Also, if you have an issue please don't create a discussion next time, create a GitHub issue instead :D

**Reply by madhits** [reply]
@tejasraman Thanks for the reply, Sorry I thought questions was for this, anyway I'm on rammus and its a 5537, Dell 15R-5537. I can not use kernel 6 (well it loads but there are issues with sleep & random reboots) so I've been running kernel 5.15 without issue. 

What else would you like to know? I have tried a number of framework options and nothing get me a boot up without a crash, machine still is in the middle of an update to 116 from 115. I also tried 117 USB and im going to build a 116 USB now and see how that goes.

**Reply by tejasraman** [reply]
> @tejasraman Thanks for the reply, Sorry I thought questions was for this, anyway I'm on rammus and its a 5537, Dell 15R-5537. I can not use kernel 6 (well it loads but there are issues with sleep & random reboots) so I've been running kernel 5.15 without issue. 
> 
> What else would you like to know? I have tried a number of framework options and nothing get me a boot up without a crash, machine still is in the middle of an update to 116 from 115. I also tried 117 USB and im going to build a 116 USB now and see how that goes.

Enable verbose in the framework options (it asks separately) and take a slow motion video.

If you can boot with 116 or 117, type `cat /var/log/dmesg > ~/Downloads/dmesg.txt` and send that file in Downloads to me

**Reply by madhits** [reply]
Ok will do, ill report back shortly. THANKS

**Comment by madhits** [comment]
I Tried to upload a video and its way to big instead here is a shot of the last screen that its stuck on, There is also a video but its 1 second kinda pointless. There are 2 points where the bootup freezes and one is where the sda1 is remounted and the second is where the elan touch screen is loaded up. Those are the 2 most critical points in the bootup that occur and i believe the first one is ok, 2nd one is the dead stop point. THe video wont work at all anyway, i guess that was pointless. You can see both points on the screenshot anyway, the line sda1 remounted quota mode journaled is where it stops first and stays for about 10-15mins.


https://github.com/sebanc/brunch/assets/20798443/0976caa7-1459-4e5a-9403-58dc7aac361a

![20231008_114450](https://github.com/sebanc/brunch/assets/20798443/939382aa-c434-43be-9709-78543b2f4bf7)


Could this just be an issue with the elan driver now being used in chromeos, as i said the framework option makes no difference, so maybe that option needs a new update to it.

**Reply by tejasraman** [reply]
> I Tried to upload a video and its way to big instead here is a shot of the last screen that its stuck on, There is also a video but its 1 second kinda pointless. There are 2 points where the bootup freezes and one is where the sda1 is remounted and the second is where the elan touch screen is loaded up. Those are the 2 most critical points in the bootup that occur and i believe the first one is ok, 2nd one is the dead stop point.
> 
> 
> https://github.com/sebanc/brunch/assets/20798443/0976caa7-1459-4e5a-9403-58dc7aac361a
> 
> ![20231008_114450](https://github.com/sebanc/brunch/assets/20798443/939382aa-c434-43be-9709-78543b2f4bf7)
> 

This seems to have something to do with a kernel compilation flag, but have you tested kernel 5.10? I'll see if that flag is set.to.false and submit a PR if so

**Reply by madhits** [reply]
ok ill try to boot with 5.10 kernel...

results, same error but it stops sooner..
![20231008_125117](https://github.com/sebanc/brunch/assets/20798443/4c5de604-d2eb-4e57-9302-f4e1a9bdd0d6)

OK So what is the kernel compilation flag, how do check if its set to false?

Also i just made a USB for 116 and that does boot up via USB with brunch_r117_stable.. again 117 fails at this..

So now I'm really confused as I was moving from 115 to 116 via bitedasher script and it failed to load but now 116 via USB it is loading, I guess the update somehow failed to complete. I have yet to install on the HD 116 so it still may fail when I go to do that, but I'd love to recover my download folder/files if i can and understand this better and or fix it?

**Reply by madhits** [reply]
Ok this is interesting, I tried a few framework options and got 1 additional line past where i was before (see attached), So I think there is most likely a framework option or kernel command I can pass to fix this boot up.

![20231008_134117](https://github.com/sebanc/brunch/assets/20798443/0439f53b-8cac-4d10-9344-691bc7867214)

**Reply by tejasraman** [reply]
I wouldn't recommend leaving this on, but see if it boots with the option that you have on and `selinux=0` as a kernel parameter (it asks)

**Reply by madhits** [reply]
Ill try, Does the tpm2 error have anything to do with it not booting possibly?

**Reply by tejasraman** [reply]
> Ill try, Does the tpm2 error have anything to do with it not booting possibly?

Well, that's why we're trying to disable selinux.

Also, how did you get to that state? What framework option?

**Reply by madhits** [reply]
Ok so the problem is with the tpm simulator that @sebanc is using? the selinux=0 command gives me a panic, ill upload a picture in a second.. also go back and look at the framework options again and they were native chromebook image and als option, I had no idea what those do, just trial and error.

![20231008_135643](https://github.com/sebanc/brunch/assets/20798443/d126278f-dc3f-4db8-b232-661ac329e145)

**Reply by madhits** [reply]
@tejasraman Thanks for your help again, it appears my install got corrupted it seems, that is what @sebanc said when he looked at it. What do you think? However I re-downloaded 117 brunch and 117 rammus and still it wont boot up via USB. So Although he said it was corrupt why wont 117 USB boot still? I downloaded 3 times now, no way they were also corrupted. I think there is an issue with 117 that probably requires a fix of some kind and although its a lil more work to fix my install id like to try if its not corrupted. I guess the way to know for sure at least for me (less knowledgable then sebanc, you and westera) is if selinux=0 actually does not cause a crash on a normal install of brunch, if that's the case I would know its corrupt, otherwise could it be something else. Anyway @tejasraman you can prob answer that one.

**Reply by tejasraman** [reply]
> @tejasraman Thanks for your help again, it appears my install got corrupted it seems, that is what @sebanc said when he looked at it. What do you think? However I re-downloaded 117 brunch and 117 rammus and still it wont boot up via USB. So Although he said it was corrupt why wont 117 USB boot still? I downloaded 3 times now, no way they were also corrupted. I think there is an issue with 117 that probably requires a fix of some kind and although its a lil more work to fix my install id like to try if its not corrupted. I guess the way to know for sure at least for me (less knowledgable then sebanc, you and westera) is if selinux=0 actually does not cause a crash on a normal install of brunch, if that's the case I would know its corrupt, otherwise could it be something else. Anyway @tejasraman you can prob answer that one.

Where did @sebanc answer you? And I need to test selinux myself, it was just to test it...

**Reply by madhits** [reply]
He answered on discord. He seemed to have know what it was quickly, I was surprised it was corrupted. Again seeing as brunch 117 wont boot off USB for me is still a mystery. I'm kinda hoping hes wrong but the selinux test is a good one, cuz getting a crash on that really does not make sense.

**Reply by tejasraman** [reply]
> He answered on discord. He seemed to have know what it was quickly, I was surprised it was corrupted. Again seeing as brunch 117 wont boot off USB for me is still a mystery. I'm kinda hoping hes wrong but the selinux test is a good one, cuz getting a crash on that really does not make sense.

Have you tried a different recovery image (e.g coral)? Just for troubleshooting

**Reply by madhits** [reply]
No, but 116 boots fine with the same version of brunch, ill try another image though.. Does the selinux=0 crash for you?

**Reply by tejasraman** [reply]
> No, but 116 boots fine with the same version of brunch, ill try another image though.. Does the selinux=0 crash for you? 
> 

No it doesn't.

Just try coral to rule out the GPU driver changes that were made - I've seen some issues with that

**Reply by madhits** [reply]
oK ill try Coral, However Im ok with 116 for now if I must reinstall. Since selinux does not crash for you i'm definitely corrupt, That is the answer I was looking for. A kernal crash like that is very rare without a serious problem.

**Comment by madhits** [comment]
UPDATE: in case anyone wants to know, if you copy your partition 1 with gparted and restore it with gparted you can recover your chrome os install after crash. I think this mostly works because brunch uses software TPM that and if its the same computer it works.
